addons/rose/addon/components/rose/loading-button/index.js

```diff
@@ -0,0 +1,20 @@
+import button from '../button/index';
+
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+export default class LoadingButton extends button {
+  // =actions
+  @tracked isLoading = false;
+
+  @action
+  async toggleRefresh() {
+    this.isLoading = true;
+    try {
+      await this.args.onClick();
+    } catch (e) {
+      console.error('Error while loading data', e);
+    }
+    this.isLoading = false;
```

Comment:
Do we want to wait for line 14 code before flagging `this.isLoading` as `false?`

Reply:
yes! We have to wait for the onClick to happen. `this.isLoading` is what helps us indicate the state of the button - whether it is loading or ready to reload. And, if there's an error during the onClick event, we simply log the error and set the `this.isLoading` to false, which would allow the user to `refresh` the data, or it would be stuck at the `loading` state.  Hope this answers your question.

Reply:
Yes, this answer my question.

Then we will want the `this.loading=false` wrapped within a `.then()` statement, so we wait for the promise of `await this.args.onClick();` to be resolved.

Reply:
I could be mistaken, but won't the `await` before `this.args.onClick()` prevent the code from continuing until that promise is resolved. I thought we would only need a `.then()` if this was not an `async` function.

Reply:
@cameronperera You are right. Being an `async` function behaves as you described.

@DhariniJeeva I was mistaken when reading this code. Resolving this comment and approving PR.